packages/tools/client-debugger/client-debugger/src/messaging/AudienceMessages.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IClient } from "@fluidframework/protocol-definitions";
+import { AudienceChangeLogEntry } from "../Logs";
+import { HasContainerId } from "./DebuggerMessages";
+
+import { IDebuggerMessage } from "./Messages";
+
+/**
+ * Metadata of clients within the Audience List
+ * @public
+ */
+export interface AudienceClientMetaData {
+	clientId: string;
+	client: IClient;
+}
+
+/**
+ * Inbound event requesting {@link AudienceChangeLogEntry} of the Container with the specific ID.
+ * Will result in the {@link AudienceEventMessage} message being posted.
+ *
+ * @public
+ */
+export interface GetAudienceMessage extends IDebuggerMessage<HasContainerId> {
+	type: "GET_AUDIENCE_EVENT";
+}
+
+/**
+ * Message data format used by {@link AudienceEventMessage}.
+ * @public
+ */
+export interface AudienceEventMessageData extends HasContainerId {
+	/**
+	 * Contents of the Audience event
+	 */
+	audienceState: AudienceClientMetaData[];
+	audienceHistory: readonly AudienceChangeLogEntry[];
+}
+
+/**
+ * Outbound event listing the current audience and audience history of the application
+ * AudienceState: List of AudienceClientMetaData with clientId and IClient
```

Comment:
I don't think we need to document the data properties here, since they are captured by the type system via `AudienceEventMessageData`. But we should make sure the properties are documented on the type (`audienceHistory` is currently missing docs).

Reply:
Comments were removed, and added explanation on `audienceHistory`.